### PR TITLE
[GEARPUMP-213] build docs with MkDocs

### PR DESCRIPTION
**Pull request by manuzhang**
build docs with [MkDocs](http://www.mkdocs.org/) instead of Jekyll. MkDocs has a "readthedocs" theme which supports two-level table of contents sidebar. It's easier for users to navigate through the documentation. Here is a snapshot.

snip20160923_5

### CODE REVIEW

**Comment by manuzhang** [comment]
@whjiang @kkasravi @huafengw what do you think ?

**Comment by huafengw** [comment]
I'm OK with this change.

**Comment by kkasravi** [comment]
+1 - though i haven't seen the output

File: docs/docs/deployment/deployment-msg-delivery.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+## How to deploy for At Least Once Message Delivery?
```
Comment (whjiang) [inline docs/docs/deployment/deployment-msg-delivery.md]: why this file is under docs/docs/deployment instead of one "docs"?

Comment (manuzhang) [inline docs/docs/deployment/deployment-msg-delivery.md]: inner `docs` is the default source directory used by MkDocs and all documentations are under the outer `docs` directory

**Comment by whjiang** [comment]
+1.

**Pull request closed without merging**